DOC: Fixed example section in pandas/core/dtypes/*.py

ShaharNaveh opened the pull request:
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Commit:
DOC: Fixed example section in pandas/core/dtypes/*.py

WillAyd commented on pandas/core/dtypes/common.py:
```diff
@@ -1013,11 +1016,7 @@ def is_datetime64_any_dtype(arr_or_dtype) -> bool:
     >>> is_datetime64_any_dtype(np.array([], dtype=np.datetime64))
     True
-    >>> is_datetime64_any_dtype(pd.DatetimeIndex([1, 2, 3],
-                                dtype=np.datetime64))
```
I think you can just change dtype here to `"datetime64[ns]"` - not sure on general rules for passing `np.datetime64` without a base as a dtype argument but cc @jbrockmendel

jbrockmendel replied:
yah, this will raise ValueError.  Either don't pass a dtype at all or pass "datetime64[ns]"

WillAyd commented on pandas/core/dtypes/common.py:
```diff
@@ -1053,14 +1052,9 @@ def is_datetime64_ns_dtype(arr_or_dtype) -> bool:
+    >>> is_datetime64_ns_dtype(np.array([], dtype="datetime64[ps]"))  # wrong unit
     False
-    >>> is_datetime64_ns_dtype(pd.DatetimeIndex([1, 2, 3],
-                               dtype=np.datetime64))  # has 'ns' unit
```
same comment here

WillAyd reviewed (changes requested):
Thanks for taking a look at these. Is doctest picking these up?

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-fix-pytest

Commit:
Fixes for review

REF:
https://github.com/pandas-dev/pandas/pull/31451#discussion_r373501601
https://github.com/pandas-dev/pandas/pull/31451#discussion_r373502296

jreback commented:
@datapythonista @jorisvandenbossche if any comments

WillAyd reviewed (approved):
Lgtm. If there’s a way for doctest to pick these up and automatically check would be all for it in a follow up

jorisvandenbossche commented on pandas/core/dtypes/cast.py:
```diff
@@ -1,4 +1,6 @@
-""" routings for casting """
+"""
+Routings for casting.
```
```suggestion
Routines for casting.
```

jorisvandenbossche reviewed (changes requested):
@MomIsBestFriend can you add those files to the "DOCTESTS" section in code_checks.sh? (or are not yet all docstrings of the files passing?)

ShaharNaveh committed:
Update pandas/core/dtypes/cast.py

Co-Authored-By: Joris Van den Bossche <jorisvandenbossche@gmail.com>

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-fix-pytest

Commit:
Added check in the CI as suggested

REF:
https://github.com/pandas-dev/pandas/pull/31451#pullrequestreview-352072757

MomIsBestFriend commented:
> @MomIsBestFriend can you add those files to the "DOCTESTS" section in code_checks.sh? (or are not yet all docstrings of the files passing?)

@jorisvandenbossche Yes, all the doctests are now passing, I have added it in 2a330b4

jreback reviewed (approved):
lgtm. @jorisvandenbossche

jorisvandenbossche commented:
Thanks!
